src/book/02-system/01-nubots/02-build-system.mdx

```diff
@@ -2,6 +2,472 @@
+| nbs       | For working with NBS recordings                                |
+| run       | Run a role locally                                             |
+| shell     | Exposes a shell in the docker image                            |
+| target    | Select a target to work on                                     |
+
+##### Build
+
+As one might expect, `./b build` will build all enabled roles and required dependencies using `ninja`.
+If you only wish to build a single role (e.g. `sensortest`) then use the command
+
+```bash
+./b build sensortest
+```
+
+If unit tests are enabled in the CMake configuration (`-DBUILD_TESTS=ON`) then `./b build test` can be run to execute all of the unit tests.
+
+##### Configure
+
+`./b configure` will ensure that a build folder is created in the current Docker image and will then run `cmake` on the source code ensuring that CMake creates all of the necessary build files inside of the build folder.
+
+It is possible to specify extra arguments to `cmake` to control the configuration process. For example, to enable the building of unit tests one would run
+
+```bash
+./b configure -- -DBUILD_TESTS=ON
+```
+
+Another good argument to specify is `CMAKE_BUILD_TYPE`. By default, `CMAKE_BUILD_TYPE` is set to `Debug`, while this is good while you are in active development of a module, you will get very poor (slow) runtime performance.
+To get better runtime performance change `CMAKE_BUILD_TYPE` to `Release` and then rebuild your code using the following commands
+
+```bash
+./b configure -- -DCMAKE_BUILD_TYPE=Release
+./b build
+```
+
+<Alert type="info">
+
+`--` is needed to prevent python from trying to process the argument itself. This is only needed if you are specifying arguments that start with a `-`.
+
+</Alert>
+
+Passing `-i` to `./b configure` will open up an interactive window with `ccmake` so that you may modify the CMake configuration.
+
+##### Edit
+
+If you are running roles on your local machine and you need to edit the configuration or data files for the modules in the role then the preferred method is to edit the file in the source tree and then rebuild the code using `./b build` to copy the edited file into the build folder.
+
+However, some modules generate their configuration file at build-time, making it impossible to edit the file in the source tree. In this situation, you can use `./b edit` to open an interactive window with a text editor allowing you to edit the configuration file in the build tree.
+
+One module that generates it configuration file at build time is `support::logging::DataLogging`, to edit the configuration file for `DataLogging` run
+
+```bash
+./b edit config/DataLogging.yaml
+```
+
+The `edit` command uses the editor that is defined in your host shell (using the `EDITOR` environment variable). If `EDITOR` is not set then it will default to `nano`. Currently, the only supported editors are `vim` and `nano`, with `nano` used as the fallback in all cases.
+
+##### Foot down
```

Comment:
Someone should move both this, and the code in NUbots into the nbs subcategory

Reply:
Also another PR since it is dependant on code changes in the other repo